firebase-firestore/src/main/java/com/google/firebase/firestore/remote/Datastore.java

```diff
@@ -221,6 +221,45 @@ public static boolean isPermanentError(Status status) {
     }
   }
 
+  /**
+   * Determines whether the given error code that represents a permanent error when received in
+   * response to a non-write operation.
+   *
+   * @see #isPermanentWriteError for classifying write errors.
+   */
+  public static boolean isPermanentError(FirebaseFirestoreException.Code code) {
```

Comment:
Let's not duplicate this method. 😄 

You could change the existing method to:

```
  public static boolean isPermanentError(Status status) {
    return isPermanentError(FirebaseFirestoreException.Code.fromValue(status.getCode().value()));
  }
```

Or: 
1. Change this method to accept FirebaseFirestoreException instead of Code (and just call .getCode() to get the code).
2. Remove the old method and change all callers of `Datastore.isPermanentError(status)` to `Datastore.isPermanentError(Util.exceptionFromStatus(status))`